### PR TITLE
fix #20760 staff type change in join / split measure

### LINKED ISSUE #20760: Incorrect behaviour if split / join measure containing staff type change
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

If measure contains staff type change and split (or join) measure is performed on souch measure, it behaves incorrectly.
Staff type change icon disappeares.
After undo, staff type change is in incorrect measure.

### Steps to reproduce

1. score with staff type change
2. split measure containing staff type change
3. see - sign disappeared
4. undo - see, staff type change is at wrong place (other than icon)

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/1646034/5dd62632-5ff9-4431-8b0c-6b7c877f2b17

### MuseScore Version

master

### Regression

No.

### Operating system

Ubuntu 22.04

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sammik**
Resolves: #20760 

This PR prevents joining measures which contain a staff change, presenting a dialog if it is attempted:
image

it is based on top of #19086, which should be merged first, so this PR is actually only last commit here 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Review by mike-spa — APPROVED** [review]
The code looks ok to me, but @bkunda this needs testing from you before we can merge. In particular, do check the wording of the error dialog

**Comment by RomanPudashkin** [comment]
@sammik please rebase this PR. Thanks!

**Comment by sammik** [comment]
> @sammik please rebase this PR. Thanks!

@RomanPudashkin rebase done

**Comment by oktophonie** [comment]
@bkunda This still requires approved/corrected copy before it can be merged

**Review by oktophonie — APPROVED** [review]

**Comment by bkunda** [comment]
The copy should be corrected as follows: 

> **These measures cannot be joined**
> Please remove the staff type change and retry

Also the word "measures" is currently not being translated to "bars" for UK English in this dialog. Will that perhaps change though upon merging?

**Comment by Jojo-Schmitz** [comment]
> Also the word "measures" is currently not being translated to "bars" for UK English in this dialog. Will that perhaps change though upon merging?

Not through merge itself, but after merge **and** updating the translations

File: src/notation/internal/mscoreerrorscontroller.cpp [hunk 1]
```diff
@@ -150,6 +150,10 @@ void MScoreErrorsController::checkAndShowMScoreError()
         title = muse::trc("notation", "This key signature cannot be deleted");
         message = muse::trc("notation", "Please replace it with a key signature from the palettes instead.");
         break;
+    case MsError::CANNOT_JOIN_MEASURE_STAFFTYPE_CHANGE:
+        title = trc("notation", "These measures cannot be joined");
+        message = trc("notation", "Please remove the staff type change and retry");
```
Comment (Jojo-Schmitz) [inline src/notation/internal/mscoreerrorscontroller.cpp:155]: `.` at the end of the (2nd) sentence missing...

Comment (Jojo-Schmitz) [reply src/notation/internal/mscoreerrorscontroller.cpp:155]: also the `muse::` prefix

**Comment by bkunda** [comment]
Once that period `.` gets added, this should be good to go from a copy and functionality POV.

**Comment by sammik** [comment]
Just in case, because PR title became bit misleading.

This can be joined without any problem:
No-Problem

But this is problem, so dialog appeares:
Problem1

And this also:
Problem2

User needs to remove second staff type change (and all subsequent, if present), but not one in first bar.

**Comment by sammik** [comment]
Thank You @Jojo-Schmitz 

@bkunda dot added, but please check my comment above, to be 100% sure, if ewerything is right

P.S. I also rebased, to prevent vtest failure

**Pull request merged**